pennylane_lightning/core/_serialize.py

```diff
@@ -394,3 +394,18 @@ def get_wires(operation, single_op):
             controlled_wires,
             controlled_values,
         ), uses_stateprep
+
+
+def global_phase_diagonal(par, wires, controls, control_values):
+    """Returns the diagonal of a C(GlobalPhase) operator."""
+    diag = np.ones(2 ** len(wires), dtype=np.complex128)
+    controls = np.array(controls)
+    control_values = np.array(control_values)
+    ind = np.argsort(controls)
+    controls = controls[ind[-1::-1]]
+    control_values = control_values[ind[-1::-1]]
+    idx = np.arange(2 ** len(wires), dtype=np.int64).reshape([2 for _ in wires])
+    for c, w in zip(control_values, controls):
+        idx = np.take(idx, np.array(int(c)), w)
+    diag[idx.ravel()] = np.exp(-1j * par)
+    return diag
```

Comment:
This doesn't look like a serialize-related functionality. Let's look for a better place since this file is designed to host the `QuantumScriptSerializer` class.

Reply:
This functions takes in a [pennylane.operation.Operation](https://docs.pennylane.ai/en/stable/code/api/pennylane.operation.Operation.html) and spits out a NumPy array to be consumed by the C++ layer. It is a serialization util in that sense. Where do you see this go otherwise?

Reply:
I would leave it here for the time being, but I don't mind moving it to `lightning_base.py` if you prefer. But I don't think we should create a module solely for that.

Reply:
OK. We can come back to this in the future.